src/qibolab/native.py

```diff
@@ -136,34 +199,47 @@ class NativeSequence:
 
     name: str
     pulses: List[Union[NativePulse, VirtualZPulse]] = field(default_factory=list)
+    coupler_pulses: List[CouplerPulse] = field(default_factory=list)
 
     @classmethod
-    def from_dict(cls, name, sequence, qubits):
+    def from_dict(cls, name, sequence, qubits, couplers):
         """Constructs the native sequence from the dictionaries provided in the runcard.
 
         Args:
             name (str): Name of the gate the sequence is applying.
             sequence (dict): Dictionary describing the sequence as provided in the runcard.
-            qubits (list): List of :class:`qibolab.platforms.abstract.Qubit` object for all
+            qubits (list): List of :class:`qibolab.qubits.Qubit` object for all
                 qubits in the platform. All qubits are required because the sequence may be
                 acting on qubits that the implemented gate is not targeting.
+            couplres (list): List of :class:`qibolab.couplers.Coupler` object for all
+                couplers in the platform. All couplers are required because the sequence may be
+                acting on couplers that the implemented gate is not targeting.
         """
         pulses = []
+        coupler_pulses = []
 
         # If sequence contains only one pulse dictionary, convert it into a list that can be iterated below
         if isinstance(sequence, dict):
             sequence = [sequence]
 
         for i, pulse in enumerate(sequence):
             pulse = pulse.copy()
-            qubit = qubits[pulse.pop("qubit")]
             pulse_type = pulse.pop("type")
-            if pulse_type == "virtual_z":
-                phase = pulse["phase"]
-                pulses.append(VirtualZPulse(phase, qubit))
+            if pulse_type == "coupler":
+                coupler = couplers[pulse.pop("coupler")]
+                duration = pulse["duration"]
+                amplitude = pulse["amplitude"]
+                shape = pulse["shape"]
+                relative_start = pulse["relative_start"]
+                coupler_pulses.append(CouplerPulse(duration, amplitude, shape, coupler, relative_start))
```

Comment:
Did not test but it seems that this shortcut would work here
```suggestion
                coupler_pulses.append(CouplerPulse(**pulse))
```

Reply:
Almost, it fails because I need `coupler = couplers[pulse.pop("coupler")]` because if not I would give the pulse the `coupler.name` 

I can do this:

``` py
pulse["coupler"] = couplers[pulse.pop("coupler")]
coupler_pulses.append(CouplerPulse(**pulse))
```

Reply:
Right, that is also fine with me.